pkgs/by-name/ta/tabby-terminal/package.nix

```
@@ -0,0 +1,258 @@
+
+    electronHeaders = fetchurl {
+      url = "https://www.electronjs.org/headers/v${electronPackage.version}/node-v${electronPackage.version}-headers.tar.gz";
+      hash = "sha256-D9j1wSDzartWUfDm2UrZDrgum2X+vV+DpDlKPFCtBbA=";
+    };
+
+    electronHeadersSHA = fetchurl {
+      url = "https://www.electronjs.org/headers/v${electronPackage.version}/SHASUMS256.txt";
+      hash = "sha256-ZnWVCzTwwog4kgeFAZGsM2F2mmebrpXMVMxn/K+LWlU=";
+    };
+
+    buildInputs = [
+      nodejs
+      python3
+      fontconfig
+      electronPackage
+    ];
+
+    nativeBuildInputs = [
+      git
+      yarn
+      patch-package
+      prefetch-yarn-deps
+      fixup-yarn-lock
+      (python3.withPackages (python-pkgs: [
+        python-pkgs.distutils
+        python-pkgs.gyp
+      ]))
+      makeWrapper
+      http-server
+      pkg-config
+      libsecret
+      copyDesktopItems
+      jq
+      moreutils
+    ];
+
+    patchPhase = ''
+      runHook prePatch
+
+      jq '.version = "${version}"' app/package.json | sponge app/package.json
+
+      runHook postPatch
+    '';
+
+    configurePhase =
+      ''
+        runHook preConfigure
+
+        export buildDir=$PWD
+
+        git tag ${version}
+      ''
+      # Loop through all the yarn caches and install the deps for the respective package
+      + builtins.concatStringsSep "\n" (
+        map (cache: ''
+          cd ${cache.pkg}
+          export HOME=$PWD/yarn_home
+          fixup-yarn-lock yarn.lock
+          yarn config --offline set yarn-offline-mirror ${cache.cache}
+          echo "Installing deps for ${cache.pkg}"
+          yarn install --offline --prefer-offline --frozen-lockfile --ignore-scripts --no-progress --non-interactive
+          patch-package
+          patchShebangs node_modules
+          echo "Done!"
+          cd $buildDir
+        '') yarnCaches
+      )
+      + ''
+        cd $buildDir/node_modules
+      ''
+      # Loop thought the "built in" plugins and link them to the node_modules
+      + builtins.concatStringsSep "\n" (
+        map (plugin: ''
+          ln -fs ../${plugin} ${plugin}
+        '') builtinPlugins
+      )
+      + ''
+        cd $buildDir
+
+        runHook postConfigure
+      '';
+
+    buildPhase =
+      # Start a fake (and completely unnecessary) http-server to let electron-rebuild "download" the headers and hashes
+      ''
+        runHook preBuild
+
+        mkdir -p http-cache/v${electronPackage.version}
+        cp $electronHeaders http-cache/v${electronPackage.version}/node-v${electronPackage.version}-headers.tar.gz
+        cp $electronHeadersSHA http-cache/v${electronPackage.version}/SHASUMS256.txt
+        http-server http-cache &
+      ''
+      # Rebuild all the needed packages using electron-rebuild
+      + builtins.concatStringsSep "\n" (
+        map (pkg: ''
+          yarn --offline electron-rebuild -v ${electronPackage.version} -m ${pkg} -f -d "http://localhost:8080"
+        '') rebuildPkgs
+      )
```

Use `lib.concatMapStringsSep`. Alternatively I wonder if this should just be written in Bash:

```bash
for pkg in "$rebuildPkgs[@]"; do
  yarn --offline electron-rebuild -v ${electronPackage.version} -m $pkg -f -d "http://localhost:8000"
done
```

```suggestion
      + lib.concatMapStringsSep "\n" (pkg: ''
        yarn --offline electron-rebuild -v ${electronPackage.version} -m ${pkg} -f -d "http://localhost:8080"
      '') rebuildPkgs
```